app/components/CookieKitPopup.js

```diff
@@ -0,0 +1,355 @@
+    cookieConsents.forEach((cookieConsent) => {
+      consentSettings[cookieConsent.type] = cookieConsent.checked;
+    });
+
+    this.state = {
+      consentSettings,
+      selectedLocale: "EN",
+      isShown: false,
+    };
+
+    window.addEventListener("message", this.onMessage);
+  }
+
+  componentWillUnmount() {
+    // console.log('CookieKitPopup#componentWillUnmount');
+    window.removeEventListener("message", this.onMessage);
+  }
+
+  onMessage = (event) => {
+    // console.log('CookieKitPopup#onMessage');
+    const { onLogin } = this.props;
+    const { token } = event.data;
+
+    if (token) {
+      onLogin(token);
+    }
+  }
+
+  handleLocaleChange = (locale) => {
+    // console.log('CookieKitPopup#handleLocaleChange');
+    this.setState({ selectedLocale: locale, isShown: false });
+  }
+
+  handleCookieCheck = (e, type) => {
+    // console.log('CookieKitPopup#handleCookieCheck');
+    const { consentSettings } = this.state;
+    const checked = {
+      ...consentSettings,
+    };
+
+    checked[type] = e.target.checked;
+
+    this.setState({ consentSettings: checked });
+  }
+
+  handleCheckAll = () => {
+    // console.log('CookieKitPopup#handleCheckAll');
+    const { consentSettings } = this.state;
+
+    const isAllChecked = Object.values(consentSettings).every(checked => checked);
+
+    if (isAllChecked) {
+      const noneChecked = {};
+      Object.keys(consentSettings).forEach((cookieName) => {
+        noneChecked[cookieName] = false;
+      });
+      this.setState({ consentSettings: noneChecked });
+    } else {
+      const allChecked = {};
+      Object.keys(consentSettings).forEach((cookieName) => {
+        allChecked[cookieName] = true;
+      });
+      this.setState({ consentSettings: allChecked });
+    }
+  }
+
+  handleSubmit = () => {
+    // console.log('CookieKitPopup#handleSubmit');
+    const { onSubmit } = this.props;
+    const { consentSettings } = this.state;
+
+    onSubmit(consentSettings);
+  }
+
+  renderTextMessage(JSON) {
+    // console.log('CookieKitPopup#renderTextMessage');
+    const { selectedLocale } = this.state;
+
+    switch (selectedLocale) {
+      case "EN":
+        return JSON["en-us"];
+      case "DE":
+        return JSON["de-de"] || JSON["en-us"];
+      case "ES":
+        return JSON["es-419"] || JSON["en-us"];
+      case "FR":
+        return JSON["fr-fr"] || JSON["en-us"];
+      default:
+        return JSON["en-us"];
+    }
+  }
+
+  render() {
+    // console.log('CookieKitPopup#render');
+    const {
+      companyLogo,
+      countryCode,
+      hideBrandTag,
+      isConnected,
+      onClose,
+      privacyUrl,
+      requestDataTypes,
+      termsUrl,
+      textMessage,
+    } = this.props;
+    const { consentSettings, isShown, selectedLocale } = this.state;
+    const isAuthorized = getAccessToken();
+    const targetUrl = encodeURIComponent(window.location.origin);
+
+    // TODO: Move the following to CSS. Use media query.
```

Comment:
@dhurlburtusa This code is needed for country flags. For this we use 3-rd party library and we have to set the size of the flag in JS code. See the line 212.